### DISCUSSION TITLE
EACCES not solved after resetting udev rules.

### DISCUSSION THREAD

**Discussion by 2403772980ygy** [OpenTabletDriver/OpenTabletDriver#2817] [Q&A] [answered]
OS: Linux 6.4.0-gentoo
Installed with guru repo.

Checked udev rule presence: /libs/udev/rules.d/99-opentabletdriver.rules
Making a softlink to 00-opentabletdriver.rules does not work.

hid_uclogic disabled in dracut configuration and blacklist.conf

When running otd(not as superuser) It shows that 
```log
[Detect:Info]	Searching for tablets...
[Detect:Debug]	Searching for tablet '10moon 1060N'
[Detect:Debug]	Searching for tablet 'Acepen AP 1060'
             omitted several lines
[Detect:Debug]	Searching for tablet 'Huion Q11K'
[Detect:Debug]	Searching for tablet 'Huion Q620M'
[Device:Debug]	Initializing device 'Huion Tablet_Q620M' /sys/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3:1.0/0003:256C:006D.0001/hidraw/hidraw0
[Device:Debug]	Using report parser type 'OpenTabletDriver.Configurations.Parsers.Huion.GianoReportParser'
[Device:Debug]	Initialized string index 200
[Detect:Info]	Found tablet 'Huion Q620M'
[Detect:Debug]	Searching for tablet 'Huion WH1409 V2'
            omitted several lines
[Detect:Debug]	Searching for tablet 'XP-Pen Star G960S'
[Evdev:Error]	Failed to initialize virtual tablet. (error code EACCES)
[Huion Q620M:Info]	Output mode: Absolute Mode
[Huion Q620M:Info]	Display area: [1920x1080@<960, 540>:0°],
[Huion Q620M:Info]	Tablet area: [266.7x165.1@<133.35, 82.55>:0°],
[Huion Q620M:Info]	Clipping: Enabled
[Huion Q620M:Info]	Ignoring reports outside area: Disabled
[Huion Q620M:Info]	Tip Binding: [Mouse Button Binding: Left]@0%
[Settings:Info]	Driver is enabled.
```

Running otd with sudo works just fine.

Extra Notes: OSU lazer runs just fine.

I fear that it's udev.conf that's messing me up. Haven't edited it myself tho.

Udev .rules file can't be uploaded, so I uploaded this:

[udev_rule.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/11895257/udev_rule.txt)

What else should I paste?

Edit: Before I use OpenTabletDriver libinput was in charge of managing my tablet.

Currently, libinput does not.

```zsh
> xinput
⎡ Virtual core pointer                    	id=2	[master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer              	id=4	[slave  pointer  (2)]
⎜   ↳ Asus Keyboard                           	id=13	[slave  pointer  (2)]
⎜   ↳ ELAN1203:00 04F3:304E Mouse             	id=14	[slave  pointer  (2)]
⎜   ↳ ELAN1203:00 04F3:304E Touchpad          	id=15	[slave  pointer  (2)]
⎣ Virtual core keyboard                   	id=3	[master keyboard (2)]
    ↳ Virtual core XTEST keyboard             	id=5	[slave  keyboard (3)]
    ↳ Power Button                            	id=9	[slave  keyboard (3)]
    ↳ Power Button                            	id=6	[slave  keyboard (3)]
    ↳ Asus Wireless Radio Control             	id=7	[slave  keyboard (3)]
    ↳ Video Bus                               	id=8	[slave  keyboard (3)]
    ↳ Asus Keyboard                           	id=18	[slave  keyboard (3)]
    ↳ Asus Keyboard                           	id=12	[slave  keyboard (3)]
    ↳ Sleep Button                            	id=10	[slave  keyboard (3)]
    ↳ Asus Keyboard                           	id=11	[slave  keyboard (3)]
    ↳ AT Translated Set 2 keyboard            	id=17	[slave  keyboard (3)]
    ↳ Asus WMI hotkeys                        	id=16	[slave  keyboard (3)]
```

Edit2: After fiddling with udev moniter:

When I run otd without sudo, udev moniter does not have any output.

When I run otd with sudo, udev moniter outputs the following:
```
KERNEL[11474.758942] add      /devices/virtual/misc/uinput (misc)
KERNEL[11474.758966] add      /module/uinput (module)
KERNEL[11474.759208] add      /devices/virtual/input/input56 (input)
KERNEL[11474.759267] add      /devices/virtual/input/input56/event13 (input)
KERNEL[11474.759283] add      /devices/virtual/input/input56/js0 (input)
KERNEL[11474.759432] add      /devices/virtual/input/input56/mouse5 (input)
UDEV  [11474.760841] add      /module/uinput (module)
UDEV  [11474.761416] add      /devices/virtual/input/input56 (input)
UDEV  [11474.762004] add      /devices/virtual/input/input56/js0 (input)
UDEV  [11474.762364] add      /devices/virtual/misc/uinput (misc)
UDEV  [11474.762662] add      /devices/virtual/input/input56/mouse5 (input)
UDEV  [11474.846429] add      /devices/virtual/input/input56/event13 (input)
```

I'm pretty sure the rules are not correct for my setup. The first lines of the rule:
```
# Dynamically generated with the OpenTabletDriver.udev tool. https://github.com/OpenTabletDriver/OpenTabletDriver
KERNEL=="uinput", SUBSYSTEM=="misc", TAG+="uaccess", OPTIONS+="static_node=uinput"
# Genius i608x
```

Edit3:
```
zcat /proc/config.gz | grep UINPUT
CONFIG_INPUT_UINPUT=m
```

**Comment by X9VoiD** [comment]
Have you already tried to manually trigger udev to apply the new rules?

```
sudo udevadm trigger
```

**Reply by 2403772980ygy** [reply]
Actually I have already rebooted twice. I guess this covers udevadm trigger?

Tried another time, but didn't take effect.

Thanks anyways. :)

**Comment by jamesbt365** [comment]
It is completely possible you dont have udev installed, make sure you have it.

**Reply by 2403772980ygy** [reply]
I have eudev installed. I don't know if it works like udev or not.

But I have to say, most things are the same. At least in such case it should be.

Migrating from openRC to systemd is difficult, and I expect this to be the last solution if possible.

At least I can tell it's (partly) working, that the permissions of /dev/hidraw0 /dev/hidraw1 /dev/hidraw6 is correct.

**Reply by Eragonfr** [reply]
eudev is udev without systemd.
It should be working in the exact same way.

But I still have the same problem… I don't know why.

**Comment by gonX** [comment]
`uaccess` udev tag needs systemd-logind afaik. If you don't have this, you'll need to change the udev rule to `chmod 0666 /dev/uniput` or similar. Or help us figure out a eudev-compatible solution.

**Comment by jamesbt365** [comment, marked as answer]
Try using the new `opentabletdriver` package instead of `opentabletdriver-bin` (its outdated and no longer actively maintained).

This will likely be fixed by this new package & with the release of newer versions of the driver, we changed the way our udev rules work and this should work better for you, else what @gonX said will likely be the answer to the problem.
